bluesky_queueserver/manager/comms.py

```diff
@@ -1,13 +1,94 @@
 import threading
 import pprint
 import json
+import asyncio
+import uuid
 from jsonrpc import JSONRPCResponseManager
 from jsonrpc.dispatcher import Dispatcher
 
 import logging
 logger = logging.getLogger(__name__)
 
 
+class CommTimeoutError(TimeoutError):
+    """
+    Raised when communication error occurs
+    """
+    pass
+
+
+class CommJsonRpcError(RuntimeError):
+    """
+    Raised when returned json-rpc message contains error
+
+    Parameters
+    ----------
+    message: str
+        Error message
+    error_code: int
+        Error code (returned by `json-rpc`)
+    error_type: str
+        Error type (returned by `json-rpc` or set to `'CommJsonRpcError'`)
+    """
+    def __init__(self, message, error_code, error_type):
+        super().__init__(message)
+        # TODO: change 'code' and 'type' to read-only properties
+        self.__error_code__ = error_code
+        self.__error_type__ = error_type
+
+    @property
+    def error_code(self):
+        return self.__error_code__
+
+    @error_code.setter
+    def error_code(self, error_code):
```

Comment:
Why over-ride the default `AttributeError` with a `RuntimeError`?

Reply:
Do you mean that we don't need the setter method here at all?

Reply:
I will merge now and do the changes in the next PR.

Reply:
Yeah, if you don't provide a setter python provides a default failure mode

```
In [2]: class Bob: 
   ...:     @property 
   ...:     def foo(self): 
   ...:         return 1 
   ...: b = Bob()                                                                                                                                 

In [3]: b.foo = 5                                                                                                                                 
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-3-b9a5dfa13086> in <module>
----> 1 b.foo = 5

AttributeError: can't set attribute
```